ui/src/app/views/recent-work/component.ts

```diff
@@ -14,21 +23,57 @@ import {
 export class RecentWorkComponent implements OnInit {
   resourceList: RecentResource[];
   fullList: RecentResource[] = [];
+  @Input('workspace')
+  workspace: Workspace;
+  accessLevel: WorkspaceAccessLevel;
   startIndex = 0;
+  cssClass: string;
+  resourcesLoading = false;
   constructor(
-    private userMetricsService: UserMetricsService
+    private userMetricsService: UserMetricsService,
+    private workspacesService: WorkspacesService,
+    private cohortsService: CohortsService,
+    private route: ActivatedRoute,
   ) {}
   index: Number;
   size = 3;
+
   ngOnInit(): void {
+    this.resourcesLoading = true;
     this.updateList();
   }
 
   updateList(): void {
-    this.userMetricsService.getUserRecentResources().subscribe((resources) => {
-      this.fullList = resources;
-      this.resourceList = this.fullList.slice(this.startIndex, this.startIndex + this.size);
-    });
+    if (this.workspace) {
+      this.cssClass = 'for-list';
+      const wsData: WorkspaceData = this.route.snapshot.data.workspace;
+      this.accessLevel = wsData.accessLevel;
+      const notebookCall = this.workspacesService
+        .getNoteBookList(this.workspace.namespace, this.workspace.id);
+      const cohortCall = this.cohortsService
+        .getCohortsInWorkspace(this.workspace.namespace, this.workspace.id);
+      Observable.forkJoin(notebookCall, cohortCall)
+        .subscribe(([notebooks, cohorts]) => {
+          const notebookResources = convertToResources(notebooks, this.workspace.namespace,
+            this.workspace.id, this.accessLevel, ResourceType.NOTEBOOK);
+          const cohortResources = convertToResources(cohorts.items, this.workspace.namespace,
+            this.workspace.id, this.accessLevel, ResourceType.COHORT);
+          this.fullList = notebookResources.concat(cohortResources);
```

Comment:
Confirming: did we want the scroll component to contain ALL resources, or just the most recent N?

Reply:
I'd assume all (for the workspace). Can't think of any reason not to? In the recent cache, (on the homepage) it makes sense to limit for a number of reasons, but here for workspace specific....

Reply:
I recall Karthik saying we didn't want the full list here but that may have been on an earlier iteration of this page. Anyways, this is good. It's easy enough to tune this later if that comes up again.